lms/templates/edxnotes/note-item.underscore

```diff
@@ -37,7 +37,9 @@
 
         <% if (tags.length > 0) { %>
             <h3 class="reference-title"><%- gettext("Tags:") %></h3>
-            <span class="reference-meta reference-tags"><%- tags.join(", ") %></span>
+            <% for (var i = 0; i < tags.length; i++) { %>
+                <a class="reference-meta reference-tags" href="#"><%- tags[i] %></a>
```

Comment:
@cahrens, I'm currently having an issue with page focus here.  Can you use the traditional link to a page fragmenthere like ```<h3 id="test_tag">test (2)</h3>``` and the ```href``` for the link would be href="#test_tag".  Right now, as a screen reader user, my focus goes to the top of the page.  I have to manually navigate to the heading associated with the tag.  Using the more standard technique of a fragment link should eliminate this.   I'm going to verify this behavior tomorrow on our windows test machine.  Apple/Safari/Voiceover is known for flaky focus tracking.

While not directly related to this PR, I was wondering if you could fix something else in this file that I noticed while testing:
There are a lot of repeating landmarks and headings.  This reduces the effectiveness of these navigation aids by polluting their respective lists with identical entries.  When I pull up a list of landmarks on any tab/view that lists Notes, I get one "References complementary" for every note on the page.  And I also get repeated non-unique Headings for Noted In, Last edited, Tags and You commented on...
Headings and landmarks should be unique. 
![repeatheadings](https://cloud.githubusercontent.com/assets/444368/7601675/c489be18-f8e7-11e4-844c-a0afe30998e6.png)
![referencescomplementary](https://cloud.githubusercontent.com/assets/444368/7601676/c48a756a-f8e7-11e4-89dd-3742cc18223f.png)
I recommend getting rid of ``` role="complementary" aria-label="<%- gettext('References') %>"``` on the ```<footer>``` element and converting all of those ```<h3>```'s to ```<p>```'s. Same goes for "You commented..."

Thanks